pymc/smc/kernels.py

```diff
@@ -186,6 +186,21 @@ def __init__(
         self.resampling_indexes = None
         self.weights = np.ones(self.draws) / self.draws
 
+        self.varlogp = self.model.varlogp
+        self.datalogp = self.model.datalogp
+
+    def initialize_rng(self, random_seed=None):
```

Comment:
Was it unnecessary to add this method? Didn't want to directly access `SMC_KERNEL.rng` since it's not initialized by just direct assignment.

Reply:
Couldn't follow, can you explain again?

Reply:
I added a new method `SMC_KERNEL.initialize_rng` which creates a new `SMC_KERNEL.rng` with given seed. It's just a convenience method for seeding the `rng` with a different seed in each worker process. Previously it wasn't necessary because the kernels were created in each process separately and `rng` is seeded during that. This PR creates the kernel before creating the worker processes and seeding has to be done separately.

I was wondering if adding a new method was unnecessary. The method doesn't do much after all. I could just do `smc.rng = np.random.default_rng(seed=random_seed)` in `_sample_smc_int` instead but I didn't want to interact with `SMC_KERNEL.rng` since I didn't see it used anywhere else outside of `SMC_KERNEL`.

Nitpicky? I agree.